package/kedro_viz/data_access/repositories/catalog.py

```diff
@@ -20,24 +21,67 @@ def get_catalog(self) -> DataCatalog:
     def set_catalog(self, value: DataCatalog):
         self._catalog = value
 
+    def get_layers(self) -> Optional[Dict]:
+        """
+        This method retrieves the layers from the catalog. For kedro-datasets versions before 1.3.0,
+        the 'layers' attribute is defined directly in the catalog.
+
+        From kedro-datasets 1.3.0 onwards, the 'layers' attribute is defined inside the 'metadata'
+        under 'kedro-viz' plugin.
+
+        Catalog before kedro-datasets 1.3.0:
+            type: pandas.CSVDataSet
+            filepath: /filepath/to/dataset
+            layers: raw
+
+        Catalog from kedro-datasets 1.3.0 onwards:
+            type: pandas.CSVDataSet
+            filepath: /filepath/to/dataset
+            metadata:
+                kedro-viz:
+                    layers: raw
+
+        Currently, Kedro up to 18.x and kedro-viz up to 6.x support both formats.
+        However, support for the old format will be discontinued from Kedro 19.x and kedro-viz
+        7.x onwards.
+        It's recommended to follow the newest format for defining layers in the catalog.
+        """
+        if self._catalog.layers:
+            return self._catalog.layers
+
+        layers: dict[str, set[str]] = defaultdict(set)
+
+        for dataset_name in self._catalog._data_sets:
+            dataset = self._catalog._get_dataset(dataset_name)
+
+            if hasattr(dataset, "metadata"):
+                metadata = dataset.metadata or {}
+                dataset_layer = metadata.get("kedro-viz", {}).get("layer")
+                if dataset_layer:
+                    layers[dataset_layer].add(dataset_name)
+        return layers or None
+
```

Comment:
This is exactly what was happening in framework side - https://github.com/kedro-org/kedro/blob/main/kedro/io/data_catalog.py#L272-L277